### ISSUE TITLE
Why do I have /data being populated when I use MySQL?

### ISSUE THREAD

**Issue by TomLewis** [Oheers/EvenMoreFish#124]
I have 1321 user files in the /data folder thats being updated every time an event happens, but I am using MySQL, so why is this data being populated?

**Comment by Oheers** [comment]
That's data about which fish each user has caught, the mysql should store number data like competitions won and the first time a fish was caught etc, having a column for every fish x a row for each user would be a pretty big database so I opted for a flatfile version instead, where each user has a list of fish and other data stored. If you look in one of the files in the /data/ folder you'll see a list of all the fish that user has caught and more data. Eventually, there'll be a front-end for this in-game with a command like /emf fish or something (it's probably unlikely to be that since /emf admin fish is too similar)

**Comment by TomLewis** [comment]
> having a column for every fish x a row for each user would be a pretty big database

You wouldnt design it like this! You make a new table so that rows are fish and one of the columns is the users ID.

Its common practise to have a table for users with ID, UUID, Username so then you can just store ID 5 for a user instaed of their entire UUID.

Using flatfiles will take up more space! 

I pick MySQL for many reasons, backups, all data in a database in one place, searcahble data, queryable data, being able to make a web frontend to dispaly infomration. All of this gets 1000x harder with flatfiles that can be easily corrupted, hard to load when you have thousands of users.

To put this into context ive used this plugin a week? two weeks? and I already have 1740 flatfiles, this will not sustain over the long term. What makes it worse is if you ever do move to MySQL fully, importing these will be much harder later the more there are.

**Comment by Oheers** [comment]
Apologies I thought I'd replied to this, 
This is how I'm going to do it in the 1.6 update, thank you ever so much for your advice, it would've stayed in flatfile forever without you. x)

**Comment by TomLewis** [comment]
Super!

**Comment by Oheers** [comment]
Oh don't worry, there'll be an /emf migrate command. I intend to start working on the re-write at some point later today.

**Comment by TomLewis** [comment]
Nope!
You already have a users table, so no need to store UUID twice! First you add a column to your `Users` table (Which youre going to re-name I hope!) that adds `ID` auto increment for each user.

So columns would be `id (Auto increment and indexed), uuid, competitions_won, fish_sold`

You then update your `Fish2` database (again needs rename and prefix) and `first_fisher` and `largest_fisher` gets the `id` of the user instead of their UUID.

Now for the /data folder, you wouldn't want each column as a fish name, because new fish can be added! You would want each fish as a row.
For example you have a UUID of each user, then you have a JSON file with say 
```
  {
    "r": "Uncommon",
    "n": "Nemo",
    "c": 35,
    "t": 1650625404,
    "l": 149.7
  },
  {
    "r": "Uncommon",
    "n": "Starfish",
    "c": 19,
    "t": 1650625468,
    "l": 146.2
  },
```

You do the same in a table, where `r, n, c, t, l` are all column names, im going to try and guess what these stand for.

So it would be like this;
`ID, Rarity, Name, C (not sure what this is), Time, Length`
Then each row would be the same as you do with JSON, but the ID is the ID of the player from the other table, and you just index the ID's in mysql so you can join them fast!

```
2, Uncommon, Nemo, 35, 1650625404, 149.7
2, Uncommon, Nemo, 35, 1650625404, 149.7
2, Uncommon, Nemo, 35, 1650625404, 149.7
50, Uncommon, Nemo, 35, 1650625404, 149.7
50, Uncommon, Nemo, 35, 1650625404, 149.7
29, Uncommon, Nemo, 35, 1650625404, 149.7
```

Hopefully this makes sense.

Im not sure if you want a separate competitions table to record each competition too?

**Comment by Oheers** [comment]
Thank you ever so much for your help, I've started work on it now and it'll be available in the next spigot release :)

**Comment by Oheers** [comment]
This has been fully implemented with the pull request on the latest commit for the master branch, set to appear in the latest v1.6 update. There won't be any way of viewing this data in-game yet though.

**Comment by sarhatabaot** [comment]
Fixed via #185
